jetty-slf4j-impl/src/main/java/org/eclipse/jetty/logging/JettyLoggerFactory.java

```diff
@@ -74,109 +79,93 @@ public Logger getLogger(String name)
             if (logger.getName().startsWith(prefix))
-            {
                 childConsumer.accept(logger);
-            }
         }
     }
 
-    public JettyLogger getRootLogger()
+    JettyLogger getRootLogger()
     {
         return rootLogger;
     }
 
     private JettyLogger createLogger(String name)
     {
-        // or is that handled by slf4j itself?
         JettyAppender appender = rootLogger.getAppender();
         int level = this.configuration.getLevel(name);
         boolean hideStacks = this.configuration.getHideStacks(name);
         return new JettyLogger(this, name, appender, level, hideStacks);
     }
 
-    /**
-     * Condenses a classname by stripping down the package name to just the first character of each package name
-     * segment.Configured
-     *
-     * <pre>
-     * Examples:
-     * "org.eclipse.jetty.test.FooTest"           = "oejt.FooTest"
-     * "org.eclipse.jetty.server.logging.LogTest" = "orjsl.LogTest"
-     * </pre>
-     *
-     * @param classname the fully qualified class name
-     * @return the condensed name
-     */
-    protected static String condensePackageString(String classname)
+    static <T> T walkParentLoggerNames(String startName, Function<String, T> nameFunction)
     {
-        if (classname == null || classname.isEmpty())
-        {
-            return "";
-        }
+        if (startName == null)
+            return null;
 
-        int rawLen = classname.length();
-        StringBuilder dense = new StringBuilder(rawLen);
-        boolean foundStart = false;
-        boolean hasPackage = false;
-        int startIdx = -1;
-        int endIdx = -1;
-        for (int i = 0; i < rawLen; i++)
+        // Checking with FQCN first, then each package segment from longest to shortest.
+        String nameSegment = startName;
+        while (nameSegment.length() > 0)
         {
-            char c = classname.charAt(i);
-            if (!foundStart)
-            {
-                foundStart = Character.isJavaIdentifierStart(c);
-                if (foundStart)
-                {
-                    if (startIdx >= 0)
-                    {
-                        dense.append(classname.charAt(startIdx));
-                        hasPackage = true;
-                    }
-                    startIdx = i;
-                }
-            }
-
-            if (foundStart)
-            {
-                if (!Character.isJavaIdentifierPart(c))
-                {
-                    foundStart = false;
-                }
-                else
-                {
-                    endIdx = i;
-                }
-            }
+            T ret = nameFunction.apply(nameSegment);
+            if (ret != null)
+                return ret;
+
+            // Trim and try again.
+            int idx = nameSegment.lastIndexOf('.');
+            if (idx >= 0)
+                nameSegment = nameSegment.substring(0, idx);
+            else
+                break;
```

Comment:
last Segment should be ROOT_LOGGER_NAME, not an empty string.

Reply:
I agree, so it was wrong in #4831 too, right?

Reply:
Yeah, I think it has been wrong since before this PR.